modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/SoapMultipartPayload.java

```diff
@@ -19,33 +22,55 @@
  *
  * @since 4.0
  */
-public class SoapMultipartPayload extends DefaultMultiPartPayload {
+public class SoapMultipartPayload implements MultiPartPayload {
+
+  private final DefaultMultiPartPayload multipart;
 
   public SoapMultipartPayload(List<Message> parts) {
-    super(parts);
+    this.multipart = new DefaultMultiPartPayload(parts);
   }
 
   /**
    * Returns the body part content of the {@link SoapMultipartPayload}.
    *
    * @return an String with the XML response.
    */
-  public String getBody() {
-    return (String) getPart(BODY_ATTRIBUTES.getName()).getPayload().getValue();
+  public InputStream getBody() {
+    return (InputStream) multipart.getPart(BODY_ATTRIBUTES.getName()).getPayload().getValue();
   }
 
   /**
    * Returns the attachments parts content of the {@link SoapMultipartPayload}.
    *
-   * @return a {@link List} with the attachments.
+   * @return a {@link Map} with the attachments.
    */
-  public Map<String, Object> getAttachments() {
-    return getParts().stream()
+  public Map<String, InputStream> getAttachments() {
+    return multipart.getParts().stream()
         .filter(part -> !getPartName(part).equals(BODY_ATTRIBUTES.getName()))
-        .collect(new ImmutableMapCollector<>(this::getPartName, part -> part.getPayload().getValue()));
+        .collect(new ImmutableMapCollector<>(this::getPartName, part -> (InputStream) part.getPayload().getValue()));
   }
 
   private String getPartName(Message part) {
     return ((PartAttributes) part.getAttributes().getValue()).getName();
   }
+
+  @Override
+  public List<Message> getParts() {
+    return multipart.getParts();
+  }
+
+  @Override
+  public List<String> getPartNames() {
+    return multipart.getPartNames();
+  }
+
+  @Override
+  public Message getPart(String partName) {
+    return multipart.getPart(partName);
```

Comment:
org.mule.runtime.core.message.DefaultMultiPartPayload#getPart does a findFirst().get(), is it possible to ask for a part that is not there? if that happens, everything blows up